### ISSUE TITLE
Cuando se importa un INP y luego ficheros Shape

### ISSUE THREAD

**Issue by Ostoveni** [neslerel/BetaTesters_QGRed#33]
He realizado una prueba donde importo un INP y se carga todo perfectamente. Luego importo un Shape de líneas y no se muestra ningún error. 
Por lo que he visto en el área de dibujo es que el Shape de líneas se añade a la información ya guardada en la estructura de datos anterior (fichero INP).

Propuesta:
------------
Mostrar un mensaje al igual que cuando importamos un INP sobre otro INP importado previamente. Lo de "añadir o completar un modelo de red con ficheros shape" debería ser una herramienta aparte de la función de importar un modelo de red, sea INP o SHP.

**Comment by nimonro** [comment]
Yo también he hecho algunas pruebas al respecto. A mi modo de ver sería útil tener la opción de opción incremental tanto desde inp como de shp. No sé si sea un problema generalizado pero yo ahora me encuentro con que tengo unos 30 modelos separados que me gustaría integrar en uno solo. Mi procedimiento ha sido importar todos los inp a formatos de QGisRed y luego ir importando los shp sobre uno de los modelos base.
Serían útiles también una herramienta que evite la duplicidad de Id's, a lo mejor añadiendo algún sufijo según de qué modelo proceda el elemento, y una herramienta de verificación de las fronteras donde es posible que haya elementos que se solapen.
Más allá de esto con un poco de paciencia sí es posible montar un modelo de conjunto añadiendo varios modelos pre existentes.

**Comment by fmartine** [comment]
Efectivamente, la importación incremental era algo ya previsto para incorporar en QGISRed, pero por ahora solo está parcialmente desarrollado para la importación de shapes y sin ninguna comprobación. Se podría extender a la importación de INP, pero es un poco más complejo. Incluso se podría alternar la importación desde INP y shapes. La importación incremental de shapes es otra opción para crear en temas aparte los elementos a añadir, utilizando las herramientas de QGIS, y después añadirlos de golpe todos al modeo. 

Una solución para diferenciar la importación incremental de la inicial sería añadir un checkbox en las pestañas de importación actuales tanto de shapes como de INP para dejar claro se si desea añadir información o sustituir la existente. Otras opciones son preguntar cada vez que se importa algo o asumir que siempre se añade, y en caso de querer reemplazar toda la red actual crear antes un nuevo proyecto vacío. En el antiguo GISRed habían dos opciones de menú diferenciadas: la importación desde cero anulando lo existente y la importación aditiva. 

La importación aditiva desde INP conllevaría lanzar de nuevo todas las herramientas de verificación de unicidad de IDs, no solo de los elementos sino también de las curvas y patrones, y comprobar que todas las referencias entre elementos son correctas (asociación de curvas, leyes de control, etc). Pueden haber además opciones contradictorias entre los diferentes INPs como el intervalo de simulación, valores por defecto, etc. Finalmente habría que comprobar que espacialmente no hay elementos superpuestos, y en tal caso, comprobar si son exactamente iguales, o si hay diferencias tratar de resolverlas o al menos reportarlas como error.

Estas comprobaciones deberían también hacerse para la importación incremental de shapes. Como se ve, no es una tarea sencilla si se quiere hacer bien.  

Vinculada a esta capacidad debería estar también la opción de particionar una redes en subredes independientes o hacer extractos para almacenarlos como proyectos diferentes, lo que sería objeto de otro 'issue'.

**Comment by Ostoveni** [comment]
> Yo también he hecho algunas pruebas al respecto. A mi modo de ver sería útil tener la opción de opción incremental tanto desde inp como de shp. No sé si sea un problema generalizado pero yo ahora me encuentro con que tengo unos 30 modelos separados que me gustaría integrar en uno solo. Mi procedimiento ha sido importar todos los inp a formatos de QGisRed y luego ir importando los shp sobre uno de los modelos base.
> Serían útiles también una herramienta que evite la duplicidad de Id's, a lo mejor añadiendo algún sufijo según de qué modelo proceda el elemento, y una herramienta de verificación de las fronteras donde es posible que haya elementos que se solapen.
> Más allá de esto con un poco de paciencia sí es posible montar un modelo de conjunto añadiendo varios modelos pre existentes.



Pienso que las opciones de importar un red en formato INP o Shape deben ir por separado. Lo normal es que empecemos trabajando
con un sólo fichero, en el caso del INP, o varios shapes.
Luego habría que preparar una serie de tools como fusionar redes (INPs, o Shapes), o preparar dentro del formulario de importar
la opción de importar múltiples redes (INPs, Shapes).

En cuanto a la opción de duplicidad de IDs, puedes descargarte esta herramienta que desarrollé en su día
https://www.researchgate.net/publication/328792362_Tool_to_rename_identifiers_Epanet
Tendrías que introducir modelo por modelo, e ir apuntando la posición final de cada elemento (nudos de caudal, embalses, etc) para poder
dar una numercación continua y no repetir los IDs.
Creo que también programe las referencias que se hacen en el modelo a los IDs. Puedes probar con dos modelos y ver si te es de ayuda.

En cuanto a la verificación de solapamientos (líneas) y repetición de nudos (superpuestos) es una herramienta que se me quedó por la mitad.

**Comment by Ostoveni** [comment]
Pienso que las opciones de importar un red en formato INP o Shape deben ir por separado. Lo normal es que empecemos trabajando con un sólo fichero, en el caso del INP, o varios shapes.
Luego habría que preparar una serie de tools: Fusionar redes (INPs, o Shapes), o preparar dentro del formulario de importar la opción de importar múltiples redes (INPs, Shapes).

En cuanto a la opción de duplicidad de IDs, puedes descargarte esta herramienta que desarrollé en su día. https://www.researchgate.net/publication/328792362_Tool_to_rename_identifiers_Epanet
Tendrías que introducir modelo por modelo, e ir apuntando la posición final de cada elemento (nudos de caudal, embalses, etc) para poder dar una numeración continua y no repetir los IDs. Creo que también programe las referencias que se hacen en el modelo a los IDs. Puedes probar con dos modelos y ver si te es de ayuda.

En cuanto a la verificación de solapamientos (líneas) y repetición de nudos (superpuestos) es una herramienta que se me quedó por la mitad. Estaba programada para comprobar en un sólo modelo INP.
